src/client/sandbox/storages/strategies.ts

```diff
@@ -0,0 +1,369 @@
+
+            let event: StorageEvent;
+
+            if (arguments.length === 0)
+                event = new (nativeMethods.StorageEvent as new () => StorageEvent)();
+            else if (arguments.length === 1)
+                event = new nativeMethods.StorageEvent(type);
+            else
+                event = new nativeMethods.StorageEvent(type, opts);
+
+            if (storedArea) {
+                nativeMethods.objectDefineProperty(event, 'storageArea', {
+                    get: () => storedArea,
+                    set: () => void 0,
+                });
+            }
+
+            return event;
+        });
+    }
+
+    private _overrideStoragesGetters () {
+        const storagesPropsOwner = this._nativeMethods.getStoragesPropsOwner(window);
+
+        // NOTE: Storage properties is located in Window.prototype in the IE11 and these are non configurable.
+        // We define descriptors from a prototype with an overridden getter on a window instance.
+        // We don't need define descriptors again if these was overridden.
+        const shouldDefineStorageProps = !this._nativeMethods.isStoragePropsLocatedInProto ||
+            !this._nativeMethods.objectHasOwnProperty.call(window, 'localStorage');
+
+        if (!shouldDefineStorageProps)
+            return;
+
+        this._nativeMethods.objectDefineProperties(window, {
+            'localStorage': createOverriddenDescriptor(storagesPropsOwner, 'localStorage', {
+                // @ts-ignore
+                getter: () => {
+                    this.localStorageProxy.unwrapProxy().setContext(window);
+
+                    return this.localStorageProxy;
+                },
+            }),
+
+            'sessionStorage': createOverriddenDescriptor(storagesPropsOwner, 'sessionStorage', {
+                // @ts-ignore
+                getter: () => {
+                    this.sessionStorageProxy.unwrapProxy().setContext(window);
+
+                    return this.sessionStorageProxy;
+                },
+            }),
+        });
+    }
+
+    private static _wrapKey (key: string): string {
+        const keyStr = String(key);
+
+        return STORAGE_PROPS.indexOf(keyStr) !== -1 ? API_KEY_PREFIX + keyStr : keyStr;
+    }
+
+    private static _unwrapKey (key: string): string {
+        return key.replace(API_KEY_PREFIX, '');
+    }
+}
+
+export class StorageSandboxProxylessStrategy implements StorageSandboxStrategy {
+    backup (): StoragesBackup {
+        return {
+            localStorage:   JSON.stringify([Object.keys(localStorage), Object.values(localStorage)]),
+            sessionStorage: JSON.stringify([Object.keys(sessionStorage), Object.values(sessionStorage)]),
+        };
+    }
+
+    clear (): void {
+        return void 0;
+    }
+
+    dispose (): void {
+        return void 0;
+    }
+
+    lock (): void {
+        return void 0;
+    }
+
+    unlock (): void {
+        return void 0;
+    }
+
+    restore (): StoragesBackup {
```

Comment:
The type for this already exists - https://github.com/DevExpress/testcafe-hammerhead/blob/master/src/typings/session.d.ts#L17.

Also, the empty storages backup should be defined as
```js
{
    localStorage:   '[[],[]]',
    sessionStorage: '[[],[]]',
 }
```

To add syntax sugar,  I propose converting interface `StoragesSnapshot` to class and the static `empty` property.
So, the updated code will look like this:
```js
restore (): StoragesSnapshot{
      return StoragesSnapshot.empty;
}
```